deployer/src/deployment/run.rs

```diff
@@ -292,25 +305,34 @@ async fn run(
     });
 
     info!("starting service");
-    let response = runtime_client.start(start_request).await.unwrap();
+    let response = runtime_client.start(start_request).await;
 
-    info!(response = ?response.into_inner(),  "start client response: ");
+    match response {
+        Ok(response) => {
+            info!(response = ?response.into_inner(),  "start client response: ");
 
-    let mut response = Err(Status::unknown("not stopped yet"));
+            let mut response = RuntimeResponse::Stop(Err(Status::unknown("not stopped yet")));
 
-    while let Ok(kill_id) = kill_recv.recv().await {
-        if kill_id == id {
-            let stop_request = tonic::Request::new(StopRequest {
-                deployment_id: id.as_bytes().to_vec(),
-                service_name: service_name.clone(),
-            });
-            response = runtime_client.stop(stop_request).await;
+            while let Ok(kill_id) = kill_recv.recv().await {
+                if kill_id == id {
+                    let stop_request = tonic::Request::new(StopRequest {
+                        deployment_id: id.as_bytes().to_vec(),
+                        service_name: service_name.clone(),
+                    });
+                    response = RuntimeResponse::Stop(runtime_client.stop(stop_request).await);
+
+                    break;
+                }
+            }
 
-            break;
+            cleanup(response);
         }
-    }
+        Err(ref status) => {
+            error!("failed to start service, status code: {}", status);
 
-    cleanup(response);
+            cleanup(RuntimeResponse::Start(response));
```

Comment:
I think this can call `start_crashed_cleanup()` directly. Then you won't have to worry about the enum

Reply:
My thinking is that tests (like `run::panic_in_main`) rely on the closure being called since they pass a sender into it. But there may very well be a better way to handle this.